frontend/src/views/SubmissionView.vue

```diff
@@ -54,9 +62,80 @@ onMounted(async () => {
   submissionStore.setNotes(notes);
   submissionStore.setPermits(permits);
   submissionStore.setPermitTypes(permitTypes);
-
   loading.value = false;
 });
+
+// Datatable filter(s)
+
+// State
+
+// Actions
+
+function ascendingComparator(a: any, b: any) {
+  return a > b ? 1 : -1;
+}
+
+function descendingComparator(a: any, b: any) {
+  return a < b ? 1 : -1;
+}
+
+const filteredDocuments = computed(() => {
+  let tempDocuments = getDocuments.value;
+  tempDocuments = tempDocuments.filter((x) => {
+    return searchTag.value ? x.filename.toLowerCase().includes(searchTag.value.toLowerCase()) : x;
+  });
+  switch (sortType.value) {
+    case 'filename': {
+      if (sortOrder.value == 1) {
+        tempDocuments = tempDocuments.sort((a, b) =>
+          ascendingComparator(a.filename.toLowerCase(), b.filename.toLowerCase())
+        );
+      } else {
+        tempDocuments = tempDocuments.sort((a, b) =>
+          descendingComparator(a.filename.toLowerCase(), b.filename.toLowerCase())
+        );
+      }
+      break;
+    }
+    case 'createdAt': {
+      if (sortOrder.value == 1) {
+        tempDocuments = tempDocuments.sort((a, b) => ascendingComparator(a.createdAt ?? 0, b.createdAt ?? 0));
+      } else {
+        tempDocuments = tempDocuments.sort((a, b) => descendingComparator(a.createdAt ?? 0, b.createdAt ?? 0));
+      }
+      break;
+    }
+    case 'filesize': {
+      if (sortOrder.value == 1) {
+        tempDocuments = tempDocuments.sort((a, b) => ascendingComparator(a.filesize, b.filesize));
+      } else {
+        tempDocuments = tempDocuments.sort((a, b) => descendingComparator(a.filesize, b.filesize));
+      }
+      break;
+    }
+    case 'mimeType': {
+      if (sortOrder.value == 1) {
+        tempDocuments = tempDocuments.sort((a, b) =>
+          ascendingComparator(a.mimeType.toLowerCase(), b.mimeType.toLowerCase())
+        );
+      } else {
+        tempDocuments = tempDocuments.sort((a, b) =>
+          descendingComparator(a.mimeType.toLowerCase(), b.mimeType.toLowerCase())
+        );
+      }
+      break;
+    }
+    case 'user': {
+      if (sortOrder.value == 1) {
+        tempDocuments = tempDocuments.sort((a, b) => ascendingComparator(a.user.toLowerCase(), b.user.toLowerCase()));
+      } else {
+        tempDocuments = tempDocuments.sort((a, b) => descendingComparator(a.user.toLowerCase(), b.user.toLowerCase()));
+      }
+      break;
+    }
+  }
+  return tempDocuments;
+});
```

Comment:
Structurally it's good. Stylistically, we could cut down on a lot of repeated code by creating one sort function that takes in the sortOrder to determine which way to sort.